src/gsUtils/gsUtils.h

```diff
@@ -138,6 +138,46 @@ inline void string_replace(std::string& str,
     }
 }
 
+/// converts a \a std::string to \a T
+template<typename T> inline T convert_to             (const std::string&);
+template<> inline         int convert_to<int>        (const std::string& s) { return stoi(s); }
+template<> inline      double convert_to<double>     (const std::string& s) { return stod(s); }
+template<> inline std::string convert_to<std::string>(const std::string& s) { return s; }
+
+/// \brief Creates a container \a T, splits and adds the std::string \a str
+/// into parts of type \a T::value_type with char \a ch as separator.
+/// T can be any container from std, that implements insert(iterator pos, const T::value_type& val);
+/// Conversions from std::String to T::value_type must be defined as
+/// TYPE convert_to<TYPE>(const std::string& s){return ...}
+/// \example
+/// std::set<index_t> s split<std::set<index_t> >("1,3,2,4,3,2,1", ','); s => {1,2,3,4}
+/// std::vector<std::string> v split<std::vector<std::string> >("ab.ba.aa", '.'); v => {"ab","ba","aa"}
+/// \tparam T STL container type, like std::vector<std::string> >, but not std::map
+/// \param str input string
+/// \param ch separator
+/// \param empty if true, adds also empty parts. default is false. Makes only sense with std::string.
+/// \return a container of T<T::value_type>
+template <class T>
+inline T split(const std::string& str, char ch, bool empty = false)
```

Comment:
Where is this used ? If it is not used too much then I do not really like it.
This function should be called "tokenize" as it is the standard name.
Templating with respect to the container type is not OK. It should be done by templating over an iterator,  and use e.g. back_inserter (http://www.cplusplus.com/reference/iterator/back_inserter/)

It would be much better to always return an std::list<T> . 
Please template it over the T = value type instead, 

The convert_to functions are also a little tricky and conflict with other things in G+Smo, for instance the gsGetReal in gsIO?gsXml.h

Reply:
> Where is this used ? If it is not used too much then I do not really like it.

At the moment it is not used - I used it in an earlier version of `gsFileManager::makeRelative`. But I thought maybe somebody else could need it.

> This function should be called "tokenize" as it is the standard name.

Oh, well, it's called `split` in [C#](https://docs.microsoft.com/en-us/dotnet/api/system.string.split?view=netframework-4.8) and [Java](https://docs.oracle.com/javase/9/docs/api/java/lang/String.html#split-java.lang.String-). Therefore I thought first on this name.

> Templating with respect to the container type is not OK.

That was the whole thing on it.

> It should be done by templating over an iterator, and use e.g. back_inserter (http://www.cplusplus.com/reference/iterator/back_inserter/)
> 
> It would be much better to always return an std::list .
> Please template it over the T = value type instead,
> 
> The convert_to functions are also a little tricky and conflict with other things in G+Smo, for instance the gsGetReal in gsIO?gsXml.h

So, I throw it out!?

Reply:
If we don't need it, I would say take it out for now..